File: core/client/game/src/commonGame/tool/DataRegister.cs
using ShineEngine;

/// <summary>
/// 数据注册
/// </summary>
public class DataRegister
{
	public DataRegister()
	{
	}

	/// <summary>
	/// 添加一个
	/// </summary>
	protected void add(DataMaker maker)
	{
		BytesControl.addDataMaker(maker);
	}

	/// <summary>
	/// 注册
	/// </summary>
	public virtual void regist()
	{

	}
}

File: core/clientLaya/game/src/commonGame/tool/DataRegister.ts
namespace Shine
{
    /** 数据注册类 */
    export class DataRegister
    {
        constructor()
        {
            
        }

        protected add(maker:DataMaker):void
        {
            BytesControl.addMaker(maker);
        }

        public regist():void
        {
            
        }
    }
}